src/MySqlConnector/MySqlClient/ConnectionPool.cs

```diff
@@ -14,6 +14,13 @@ internal sealed class ConnectionPool
 		{
 			cancellationToken.ThrowIfCancellationRequested();
 
+			// if all sessions are used, see if any have been leaked and can be recovered
+			if (m_sessionSemaphore.CurrentCount == 0)
+			{
+				GC.Collect();
+				RecoverLeakedSessions();
```

Comment:
I was under the assumption that once the connection was Garbage Collected, it's Dispose method would return it to the pool. When is that not the case?